doc/book/app_server/luajit_memprof.rst

```diff
@@ -0,0 +1,524 @@
+Can I start the profiler when the application is already running?
+
+**A**: Yes. Here is the example of code that can be inserted in the Tarantool
+console for a running instance.
+
+..  code-block:: lua
+    :linenos:
+
+    local fiber = require "fiber"
+    local log = require "log"
+
+    fiber.create(function()
+      fiber.name("memprof")
+
+      collectgarbage() -- Collect all objects already dead
+      log.warn("start of profile")
+
+      local st, err = misc.memprof.start(FILENAME)
+      if not st then
+        log.error("failed to start profiler: %s", err)
+      end
+
+      fiber.sleep(TIME)
+
+      collectgarbage()
+      st, err = misc.memprof.stop()
+
+      if not st then
+        log.error("profiler on stop error: %s", err)
+      end
+
+      log.warn("end of profile")
+    end)
+
+where
+
+*   ``FILENAME``—a name of the report file in binary format
+*   ``TIME``—duration of profiling, seconds.
+
+Also, you can directly call ``misc.memprof.start()`` and ``misc.memprof.stop()``
+from a console.
+
+.. _profiler_analysis:
+
+Report analysis example
+-----------------------
+
+In the example below, the following Lua code named ``format_concat.lua`` is
+investigated with the help of the memory profiler reports.
+
+..  code-block:: lua
+    :linenos:
+
+    jit.off() -- Prevent allocations on new traces.
+
+    local function concat(a)
+      local nstr = a.."a"
+      return nstr
+    end
+
+    local function format(a)
+      local nstr = string.format("%sa", a)
+      return nstr
+    end
+
+    collectgarbage() -- Clean up.
+
+    local binfile = "/tmp/memprof_"..(arg[0]):match("([^/]*).lua")..".bin"
+
+    local st, err = misc.memprof.start(binfile)
+    assert(st, err)
+
+    -- Payload.
+    for i = 1, 10000 do
+      local f = format(i)
+      local c = concat(i)
+    end
+    collectgarbage() -- Clean up.
+
+    local st, err = misc.memprof.stop()
+    assert(st, err)
+
+    os.exit()
+
+When you run this code :ref:`under Tarantool <profiler_usage_generate>` and
+then :ref:`parse <profiler_usage_parse_command>` the binary memory profile,
+you will get the following profiling report:
+
+..  code-block:: console
+
+    ALLOCATIONS
+    @format_concat.lua:8, line 9: 19998     624322  0
+    INTERNAL: 1     65536   0
+
+    REALLOCATIONS
+
+    DEALLOCATIONS
+    INTERNAL: 19998 0       558816
+            Overrides:
+                    @format_concat.lua:8, line 9
+
+    @format_concat.lua:8, line 9: 2 0       98304
+            Overrides:
+                    @format_concat.lua:8, line 9
+
+The reasonable questions regarding the report can be:
+
+*   Why are there no allocations related to the ``concat()`` function?
+*   Why the amount of allocations is not a round number?
+*   Why are there approximately 20K allocations instead of 10K?
+
+First of all, LuaJIT doesn't create a new string if the string with the same
+payload exists. This is called the string interning. So, when the string is
```

Comment:
It's worth to mention the following links:
* Wikipedia for [string interning](https://en.wikipedia.org/wiki/String_interning)
* [Lua users wiki](http://lua-users.org/wiki/ImmutableObjects), but unfortunately there is no direct anchor to the section where immutability of strings is described

So, you can provide Lua-related one within the first sentence (when you're talking about LuaJIT) and the one to Wikipedia within the second sentence (where you introduce the term).

Reply:
done